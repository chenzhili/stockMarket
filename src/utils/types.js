function judge(type){
    return (value)=>{
        type = `${type.slice(0,1).toUpperCase()}${type.slice(1)}`
        return Object.prototype.toString.call(value) === `[object ${type}]`
    }
}
export const isString = judge("String");
export const isNumber = judge("Number");
export const isBoolean = judge("Boolean");
export const isArray = judge("Array");
export const isObject = judge("Object");
export const isFunction = judge("Function");
export const isNull = judge("Null");
export const isUndefined = judge("Undefined");