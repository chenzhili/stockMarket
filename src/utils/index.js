import {pcOrH5} from "../enums/device"


/* 判断 当前 是 pc 还是 h5 */
export function browserRedirect() {
    var sUserAgent = navigator.userAgent.toLowerCase();
    if (/ipad|iphone|midp|rv:1.2.3.4|ucweb|android|windows ce|windows mobile/.test(sUserAgent)) {
        return pcOrH5.h5
    } else {
        return pcOrH5.pc
    }
}