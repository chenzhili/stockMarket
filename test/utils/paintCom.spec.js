import { paintLine } from '../../src/utils/paintCom'

let ctx = null;
beforeEach(() => {
    ctx = document.createElement('canvas').getContext('2d');
})
it('patinCom canvas mock', () => {
    // console.log(ctx);
    expect(() => ctx.arc(1, 2, 3, 4)).toThrow(TypeError);
})
describe('测试 canvas的 返回 对应的 snapshot', () => {
    let calls;
    console.log(ctx)
    describe.each`
        sx    | sy      | ex    | ey
        ${0}  |  ${0}   | ${0}  | ${0}
        ${0}  |  ${-1}  | ${-2} | ${0}
        ${0}  |  ${null}| ${0}  | ${100}
    `('参数值 sx:$sx sy:$sy ex:$ex ey:$ey', ({ sx, sy, ex, ey }) => {
        paintLine({ ctx, sx, sy, ex, ey });
        calls = ctx.__getDrawCalls();
        expect(path).toMatchSnapshot();
    })
})
