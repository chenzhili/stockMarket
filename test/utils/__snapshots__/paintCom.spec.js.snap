// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 canvas的 返回 对应的 snapshot 测试paintLine 函数 sx: sy: ex: ey: 测试 canvas 绘制是否顺利 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": 0,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;

exports[`测试 canvas的 返回 对应的 snapshot 测试paintLine 函数 sx:0 sy:-1 ex:-2 ey:0 测试 canvas 绘制是否顺利 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": -1,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": -2,
            "y": 0,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;

exports[`测试 canvas的 返回 对应的 snapshot 测试paintLine 函数 sx:0 sy:0 ex:0 ey:0 测试 canvas 绘制是否顺利 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": 0,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": 0,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;

exports[`测试 canvas的 返回 对应的 snapshot 测试paintLine 函数 sx:0 sy:null ex:0 ey:100 测试 canvas 绘制是否顺利 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": 0,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "moveTo",
        },
        Object {
          "props": Object {
            "x": 0,
            "y": 100,
          },
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "lineTo",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;

exports[`测试 canvas的 返回 对应的 snapshot 测试paintLine 函数 sx:undefined sy:undefined ex:undefined ey:undefined 测试 canvas 绘制是否顺利 1`] = `
Array [
  Object {
    "props": Object {
      "path": Array [
        Object {
          "props": Object {},
          "transform": Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ],
          "type": "beginPath",
        },
      ],
    },
    "transform": Array [
      1,
      0,
      0,
      1,
      0,
      0,
    ],
    "type": "stroke",
  },
]
`;
